ui/app/AppLayouts/Wallet/services/dapps/helpers.js

```diff
@@ -68,7 +68,7 @@ function buildSupportedNamespaces(chainIds, addresses, methods) {
         "eip155":{
             "chains": [${eipChainIds.join(',')}],
             "methods": [${methodsStr}],
-            "events": ["accountsChanged", "chainChanged"],
+            "events": ["chainChanged","accountsChanged","message","disconnect","connect"],
```

Comment:
@saledjenic @clauxx Do you know if we'll need additional implementations in the integration layer for these extra events? Can't find anything meaningful in docs, integration or testing.

Reply:
do you mean the "chainChanged" and "accountsChanged" events? we don't use them as of know from what I know, and doesn't seem like we need them, since we agree ahead of time with the dapp on the account/chains. e.g. whenever the dApp changes the chain, we don't need to react to that.

Reply:
> whenever the dApp changes the chain, we don't need to react to that

Really? Didn't know that was our intention.

>  Do you know if we'll need additional implementations in the integration layer for these extra events?

When the dapp asks for supported namespaces it actually asks if our wallet supports all events/methods that the dapp needs for functionating. In order to connect to the dapp we should at least support "required" events/methods, otherwise that dapp won't be successfully connected to our wallet. 

Now if we extend namespace we want to support, we have to add necessary listeners for those events (or connect them to appropriate handlers if already added) and also extend functionality with newly added methods to support all that the namespaces said.

At least that's how I see it.

Reply:
@saledjenic Looks like we're being called through methods and we're supposed to call WC APIs through events. https://docs.walletconnect.com/walletkit/web/usage#emitting-session-events

IDK why Paraswap requires us to send all these events, but I think it's up to us to decide when we should send these events. The difficulty I see is to find the appropriate `when` to send these events.

Reply:
Yes, that's how it should be, we have to listen for events sent from the dapp and react to them and notify the dapp about the change on the wallet side.

During the negotiation process, when we're building a namespace, we should always respond with all chains/events/actions we're supporting in our wallet.

Reply:
Makes sense! I'll add the events to enable the connection and add another task to go deeper in each signal. Looks like we're missing more handlers, not just these here and it's probably worth it's own space.

Reply:
https://github.com/status-im/status-desktop/issues/16180